src/components/InventoryModal/InventoryModal.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import { Button, Modal, ModalContent } from 'semantic-ui-react'
+import AddItem from '../AddItem'
+import SoldItem from '../SoldItem'
+import ModifyItem from '../ModifyItem'
+import RemoveItem from '../RemoveItem'
+import '../../styles/modal.css'
+
+const modalTypes = {
+  add: {
+    modalButtonText: 'Add Item',
+    modalHeaderText: 'Add Item',
+    modalColor: 'blue',
+    modalIcon: 'plus',
+    modalActionButtonText: 'Add to Inventory',
+    modalContent: <AddItem />,
+    modalCSS: 'add-item-modal-header',
+  },
+  sold: {
+    modalButtonText: 'Mark Sold',
+    modalHeaderText: 'Mark Sold',
+    modalColor: 'green',
+    modalIcon: 'checkmark',
+    modalActionButtonText: 'Mark as Sold',
+    modalContent: <SoldItem />,
+    modalCSS: 'sold-item-modal-header',
+  },
+  modify: {
+    modalButtonText: 'Edit',
+    modalHeaderText: 'Edit Item',
+    modalColor: 'yellow',
+    modalIcon: 'edit',
+    modalActionButtonText: 'Edit',
+    modalContent: <ModifyItem />,
+    modalCSS: 'modify-item-modal-header',
+  },
+  remove: {
+    modalButtonText: 'Remove',
+    modalHeaderText: 'Remove Item',
+    modalColor: 'red',
+    modalIcon: 'trash',
+    modalActionButtonText: 'Yes, remove',
+    modalContent: <RemoveItem />,
+    modalCSS: 'remove-item-modal-header',
+  },
+}
+
+const InventoryModal = ({ type }) => {
+  const [open, isModalOpen] = React.useState(false)
+  const {
+    modalButtonText,
+    modalHeaderText,
+    modalColor,
+    modalIcon,
+    modalActionButtonText,
+    modalContent,
+    modalCSS,
+  } = modalTypes[type]
+  return (
+    <Modal
+      onClose={() => isModalOpen(false)}
+      onOpen={() => isModalOpen(true)}
+      open={open}
+      trigger={
+        <Button inverted color={modalColor}>
+          {modalButtonText}
+        </Button>
+      }
+    >
+      <Modal.Header id={modalCSS}>{modalHeaderText}</Modal.Header>
+
+      <ModalContent>{modalContent}</ModalContent>
```

Comment:
Cool trick. We can replace this with something called "children". So something like... "<ModalContent>{children}</ModalContent>". Make sure to add children as one of the parameters. 

So make this work you would just pass all your form content in the different modals you've created like so...

`<InventoryModal> <Form>
      <Form.Group>
        <Form.Input
          required
          label="Name"
          placeholder="Air Jordan 1 High Mocha"
          width={XLARGE}
        /> 
.........
</InventoryModal>`

React will automatically take all those items you've passed and assign that implicitly to children. 

You can read more about it here: https://medium.com/@martin.crabtree/react-js-using-children-props-c83d5b259756

Reply:
As for the other items in the config, you can pass them in as props instead when you're calling the component.